### PR TITLE
Microsoft dcu bots

### PR DESCRIPTION
**Pull request by Phantasus**
This PR merges code for the Microsoft DCU Collector and Parser bot.

It also fixes the links in setup.py for docs files.
It adds a not yet tested proxy setup code for the collector.

It also incoporates some naming changes to the bot.

### CODE REVIEW

**Comment by aaronkaplan** [comment]
On Jun 3, 2015, at 5:57 PM, Josef P. Bernhart <notifications@github.com> wrote:

> This PR merges code for the Microsoft DCU Collector and Parser bot.
> 
> It also fixes the links in setup.py for docs files.
> It adds a not yet tested proxy setup code for the collector.
> 
> It also incoporates some naming changes to the bot.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/certtools/intelmq/pull/211

Tested it... it seems that the collector does not even start when you try to start it manually via python -m...

something is still very broken.

>

**Comment by aaronkaplan** [comment]
On Jun 3, 2015, at 5:57 PM, Josef P. Bernhart <notifications@github.com> wrote:

> This PR merges code for the Microsoft DCU Collector and Parser bot.
> 
> It also fixes the links in setup.py for docs files.
> It adds a not yet tested proxy setup code for the collector.
> 
> It also incoporates some naming changes to the bot.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/certtools/intelmq/pull/211
> 


okay, I found the bug.

So first of all, the self.parameters.proxy* settings do not exist inside the process() function.
So that would not work.

Secondly, the assignment of proxy_options.port was wrong: you assigned to proxy_options.port the value of proxy_options.hostname . Can't work.

So if these changes are done, the collector will actually fetch data.

However, there seems to be still another problem with the parser:

I am getting error messages in the form in the parser's logfile:

```
0\t\t\t\t\t\t\tN\t91:D7:30:18??\t\t\t\r\nSinkHoleMessage\t130366076981575285\tB68-1-64\t10.1.1.2\t5520\tAS25255\t10.1.1.224\t16470\tIEI2OC1sb2c6MTM5MjEzNDA5Nwk3Ny4xMTkuMTMzLjI1MQkxNjguNjMuMTg0LjIyNAk1NTIwCTE2NDcwCUI2OC0xLTY0CTAJMAkJCQkJCQlOCTlFOjE3OkQ2OkQ4Pz8=\tAT\t\t\t\t47.3333\t13.3333\t0\t0\t\t\t\t\t\t\tN\t9E:17:D6:D8??\t\t\t\r\n'
                                                                                                                                                                     2015-06-03 20:41:07,367 - microsoft-dcu-parser - ERROR - Check the following exception:
                                                                       Traceback (most recent call last):
                                                                                                           File "intelmq/lib/bot.py", line 69, in start
                                                                                                                                                           self.process()
                                                                                                                                                                           File "/home/aaron/intelmq/intelmq/bots/parsers/microsoft_dcu/parser.py", line 23, in process
                                                                                      for row in rows:
                                                                                                        File "/usr/lib/python2.7/csv.py", line 104, in next
                                                                                                                                                               row = self.reader.next()
  Error: newline inside string
                              2015-06-03 20:41:07,368 - microsoft-dcu-parser - ERROR - Pipeline connection failed (Error('newline inside string',))
                                                                                                                                                   2015-06-03 20:41:07,368 - microsoft-dcu-parser - INFO - Pipeline will reconnect in 30 seconds
                                                           2015-06-03 20:41:37,397 - microsoft-dcu-parser - INFO - Connecting to source pipeline
                                                                                                                                                2015-06-03 20:41:37,398 - microsoft-dcu-parser - INFO - Connected to source pipeline
```

(sorry for the formatting)

File: intelmq/bots/collectors/microsoft_dcu/collector.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def process(self):
+        account_name = self.parameters.azure_account_name
+        account_key = self.parameters.azure_account_key
+
+        blob_service = BlobService(account_name, account_key, protocol="https")
+        proxy_options = urlparse(self.parameters.https_proxy or "")
+
+        if proxy_options.port and proxy_options.hostname:
+            blob_service.set_proxy(host=proxy_options.hostname, port=proxy_options.hostname)
```
Comment (aaronkaplan) [inline intelmq/bots/collectors/microsoft_dcu/collector.py]: here we have a problem: port = proxy_options.hostname ... that can't work.
Also please note that the proxy_options object contains "None:None" at this stage - always.

Comment (Phantasus) [inline intelmq/bots/collectors/microsoft_dcu/collector.py]: *sigh* I hate typos.

**Comment by aaronkaplan** [comment]
merged, added a README.md file and we should be good. It's tested. Thanks!

**Comment by SYNchroACK** [comment]
Please remove the following files from the pull req:

```
 .gitignore
intelmq/lib/bot.py
intelmq/tests/__init__.py
intelmq/tests/bots/__init__.py
intelmq/tests/bots/parsers/__init__.py
intelmq/tests/bots/parsers/dcu/__init__.py
+83 −0   intelmq/tests/bots/parsers/dcu/lib.py
setup.py
```

**Comment by aaronkaplan** [comment]
On Jun 8, 2015, at 5:28 PM, Tomás Lima <notifications@github.com> wrote:

> Please remove the following files from the pull req:
> 
>  .gitignore
> intelmq/lib/bot.py
> intelmq/tests/__init__.py
> intelmq/tests/bots/__init__.py
> intelmq/tests/bots/parsers/__init__.py
> intelmq/tests/bots/parsers/dcu/__init__.py
> +83 −0   intelmq/tests/bots/parsers/dcu/lib.py
> setup.py

oh damn, okay these slipped in.
So you want to see a dedicated PR for those?

**Comment by SYNchroACK** [comment]
of course. :/

**Pull request merged**